src/plugins/intel_cpu/src/nodes/sparse_fill_empty_rows.cpp

```diff
@@ -0,0 +1,153 @@
+    } else {
+        const auto* denseShapePtr = getSrcDataAtPortAs<const int64_t>(1);
+        numRows = denseShapePtr[0];
+    }
+
+    std::unordered_set<int64_t> existingRows;
+    size_t indicesCount = indicesShape.getElementsCount() / 2;  // Divide by 2 because indices is [M, 2]
+
+    if (indicesPrecision == ov::element::i32) {
+        const auto* indicesPtr = getSrcDataAtPortAs<const int32_t>(2);
+        for (size_t i = 0; i < indicesCount; i++) {
+            existingRows.insert(static_cast<int64_t>(indicesPtr[i * 2]));
+        }
+    } else {
+        const auto* indicesPtr = getSrcDataAtPortAs<const int64_t>(2);
+        for (size_t i = 0; i < indicesCount; i++) {
+            existingRows.insert(indicesPtr[i * 2]);
+        }
+    }
+
+    size_t emptyRowsCount = numRows - existingRows.size();
+    size_t valuesCount = valuesShape.getElementsCount();
+    ov::Shape outputIndicesShape{valuesCount + emptyRowsCount, 2};
+    ov::Shape outputValuesShape{valuesCount + emptyRowsCount};
+    ov::Shape emptyRowIndicatorShape{static_cast<size_t>(numRows)};
+
+    redefineOutputMemory({outputIndicesShape, outputValuesShape, emptyRowIndicatorShape});
+    execute(strm);
+}
+
+namespace {
+struct SparseFillEmptyRowsContext {
+    SparseFillEmptyRows& node;
+};
+}  // namespace
+
+template <typename T, typename T_IND>
+void SparseFillEmptyRows::executeImpl() {
+    ov::reference::sparse_fill_empty_rows(getSrcDataAtPortAs<const T>(0),                        // values
+                                          getSrcMemoryAtPort(0)->getShape().getElementsCount(),  // values_size
+                                          getSrcDataAtPortAs<const T_IND>(1),                    // dense_shape
+                                          getSrcDataAtPortAs<const T_IND>(2),                    // indices
+                                          *getSrcDataAtPortAs<const T>(3),                       // default_value
+                                          getDstDataAtPortAs<T_IND>(0),                          // output_indices
+                                          getDstDataAtPortAs<T>(1),                              // output_values
+                                          getDstDataAtPortAs<bool>(2));                          // empty_row_indicator
+}
+
+template <typename T>
+struct SparseFillEmptyRows::SparseFillEmptyRowsExecute {
+    void operator()(SparseFillEmptyRowsContext& ctx) {
+        auto indicesPrecision = ctx.node.getParentEdgeAt(2)->getMemory().getDesc().getPrecision();
+        if (indicesPrecision == ov::element::i32) {
+            ctx.node.executeImpl<T, int32_t>();
+        } else if (indicesPrecision == ov::element::i64) {
+            ctx.node.executeImpl<T, int64_t>();
+        } else {
+            OPENVINO_THROW("SparseFillEmptyRows operation supports only i32 or i64 indices precision");
+        }
+    }
+};
+
+void SparseFillEmptyRows::execute(const dnnl::stream& strm) {
```

Comment:
```suggestion
void SparseFillEmptyRows::execute(const dnnl::stream&) {
```
looks unused

Reply:
That will be fixed with the merge of CPU PR, I'm waiting for the last review. After that I'll merge master here and only the TF FE changes will be left.